File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/common/FooterPage.java
package com.gu.fronts.integration.test.page.common;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.nwfront.NetworkFrontPage;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class FooterPage extends AbstractParentPage {

    @FindByTestAttribute(using = "copyright")
    private WebElement copyright;

    @FindByTestAttribute(using = "logo_footer")
    private WebElement logo;

    public FooterPage(WebDriver webDriver) {
        super(webDriver);
    }

    public FooterPage isDisplayed() {
        super.assertExists(copyright, logo);
        return this;
    }

    public NetworkFrontPage clickLogo() {
        logo.click();
        return pageFactory.initPage(webDriver, NetworkFrontPage.class);
    }

}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/common/FooterPage.scala
package com.gu.fronts.integration.test.page.common

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute
import com.gu.fronts.integration.test.page.nwfront.NetworkFrontPage
import com.gu.fronts.integration.test.page.util.FrontsParentPage
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed

class FooterPage(webDriver: WebDriver) extends FrontsParentPage(webDriver) {

  @FindByTestAttribute(using = "copyright")
  private var copyright: WebElement = _

  @FindByTestAttribute(using = "logo_footer")
  private var logo: WebElement = _

  def isDisplayed(): Boolean = {
    existsAndDisplayed(copyright, logo)
  }

  def clickLogo(): NetworkFrontPage = {
    logo.click()
    pageFactory.initPage(webDriver, classOf[NetworkFrontPage])
  }
}